x-pack/plugins/rule_registry/server/rule_data_plugin_service/rule_data_plugin_service.ts

```diff
@@ -195,7 +200,8 @@ export class RuleDataService implements IRuleDataService {
     return new RuleDataClient({
       indexInfo,
       resourceInstaller: this.resourceInstaller,
-      isWriteEnabled: this.isWriteEnabled(),
+      isWriteEnabled:
+        this.isWriteEnabled() && !this.isRegistrationContextDisabled(registrationContext),
```

Comment:
@fkanout Great question, yep you are right. In the beginning I thought to add a separate key, but that means I would need to change more places in the code (places where isWriteEnabled is called, I should check for registrationContexts as well). 

@weltenwort is there a benefit of combining isWriteEnabled and disabledRegistrationContexts into one key vs splitting it into separate keys?

Reply:
Maybe we can try to answer the opposite question: Why would the `RuleDataClient` need to know about the reason for being disabled? In the spirit of keeping the coupling as loose as possible it would make sense not to introduce the knowledge about a per-registration-context disablement feature out of the `RuleDataClient` if not necessary.

So what would be the benefit of passing them separately?

Reply:
@weltenwort Ok if we want to keep the coupling as loose as possible, it makes sense to keep the knowledge about per-registration-context disabling out of the RuleDataClient. @fkanout do you have any objections?

Reply:
I think the knowledge that `RuleDataClient` already has via `isWriteEnabled` is similar to `isRegistrationContextDisabled `, the difference is only the granularity of that info. (everything/selected things). Unless if `RuleDataClient` shouldn't know about `isWriteEnabled` in the first place - That, I don't know.

My understanding is once Alert-as-Data is adopted across our products, the general `xpack.rule_registry.write.enabled` will be deprecated. While the `xpack.ruleRegistry.write.disabledRegistrationContexts` could stay for long. From that point, I would say they should be separated.

Reply:
@fkanout What if we add even more granularity and disable writing per rule type id? Would we want to pass one more key for example disabledRuleTypeIds? Does the RuleDataClient need to know this level of detail? I don't know what is the correct answer.

For now I would keep it as it is. We could discuss it further if you want and come up with the most appropriate solution.

Reply:
IMHO, the only thing that RuleDataClient should know is if isWriteEnabled true or false. All the complex logics in it, makes it hard to read/understand. 

I would implement all the logic in `this.isWriteEnabled()` method at line 118.

So, it would be:
```
public isWriteEnabled(): boolean {
    return this.options.isWriteEnabled && 
    !this.options.disabledRegistrationContexts.includes(registrationContext);
  }
```

Reply:
@ersin-erdal Sounds good to me. Let me refactor and push the changes.

Reply:
> I would implement all the logic in this.isWriteEnabled() method at line 118.
It could be more readable, but it is still the same solution. We have a consensus about it, so let's merge it like that 👍🏻